### ISSUE TITLE
A URL (or any other text containing `//`) incorrectly sets `is_comment` to true

### ISSUE THREAD

**Issue by carols10cents** [rust-lang-nursery/rustfmt#2268]
## What happened 

I have a project containing this constant:

```rust
const RFC_CRATE_FILE_EXAMPLE: &[u8] = br#"{"name":"my_serde","vers":"1.0.11","deps":[{"name":"serde","req":"^1.0","registry":"https://github.com/rust-lang/crates.io-index","features":[],"optional":true,"default_features":true,"target":null,"kind":"normal"}],"cksum":"f7726f29ddf9731b17ff113c461e362c381d9d69433f79de4f3dd572488823e9","features":{"default":["std"],"derive":["serde_derive"],"std":[]},"yanked":false}
"#;
```

Using rustfmt-nightly 0.2.7 and Rust nightly 2017-11-07, if I run:

```
cargo +nightly-2017-11-07 fmt
```

I get this output from rustfmt:

```
error: line exceeded maximum width (maximum: 100, found: 414)
 --> /Users/carolnichols/rust/rustfmt-test/src/lib.rs:1
  |
1 | const RFC_CRATE_FILE_EXAMPLE: &[u8] = br#"{"name":"my_serde","vers":"1.0.11","deps":[{"name":"serde","req":"^1.0","registry":"https://github.com/rust-lang/crates.io-index","features":[],"optional":true,"default_features":true,"target":null,"kind":"normal"}],"cksum":"f7726f29ddf9731b17ff113c461e362c381d9d69433f79de4f3dd572488823e9","features":{"default":["std"],"derive":["serde_derive"],"std":[]},"yanked":false}
  |                                                                                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  = note: use `error_on_line_overflow_comments = false` to suppress the warning against line comments

warning: rustfmt may have failed to format. See previous 1 errors.
```

It says to suppress this warning, I should use `error_on_line_overflow_comments = false`, but this line isn't a comment.

## What I expected

I expected rustfmt to tell me this line is too long, but instead of telling me to use `error_on_line_overflow_comments = false`, I expected it to tell me to use `error_on_line_overflow = false` since this line isn't a comment.

## Likely cause

I believe this is because of the `//` that appears in the URL in my raw byte string, and the line length code isn't actually parsing the Rust code, so it [sets `is_comment` to true](https://github.com/rust-lang-nursery/rustfmt/blob/68b43b2240fe03edab4d0951a1d4c9bb04b559f4/src/lib.rs#L445-L447) if it sees two `/` characters in a row anywhere on a line.

**Comment by topecongiro** [comment]
This should be closed via #2257.

**Comment by carols10cents** [comment]
Awesome, thanks!!!!!
